src/dotnet-install.sh

```diff
@@ -1074,6 +1109,11 @@ do
             echo "  --architecture <ARCHITECTURE>      Architecture of dotnet binaries to be installed, Defaults to \`$architecture\`."
             echo "      --arch,-Architecture,-Arch"
             echo "          Possible values: x64, arm, and arm64"
+            echo "  --os <system>                    Specifies operating system to be used when selecting the installer."
+            echo "          Overrides the OS determination approach used by the script. Supported values: osx, linux, linux-musl, freebsd, rhel.6."  
```

Comment:
This list doesn't include supported OS for legacy links. As mentioned in 1st comment I don't want to mix them, as it increases chance of error. 
My suggestion for those who still on legacy versions to continue to use --runtime-id to override OS if needed.

Reply:
That sounds reasonable to me.